### PR TITLE
feat(server): Add instance identifier when logging in cloud

### PR DESCRIPTION
**Pull request by nalanj**
When reviewing logs in Render right now it's difficult to know what instance of server logged what. This adds a random identifier at startup that will help us filter logs to specific instances.

I thought about using hostnames but the hostnames on render are super long and not that helpful, and I'm not 100% sure they're guaranteed to be unique on their infra.

### CODE REVIEW

**Review by TBonnin — APPROVED** [review]
good idea. How does it look like in the logs?

File: packages/utils/lib/logger.ts [hunk 1]
```diff
@@ -41,10 +42,12 @@ if (!isCloud && !isEnterprise) {
         })
     ];
 } else {
+    const instanceId = isCloud ? nanoid(10) : '';
```
Comment (TBonnin) [inline packages/utils/lib/logger.ts]: we can go smaller I think. 4? 6?

File: packages/utils/lib/logger.ts [hunk 2]
```diff
@@ -41,10 +42,12 @@ if (!isCloud && !isEnterprise) {
...
+
     formatters = [
         winston.format.printf((info) => {
             const splat = info[SPLAT] && info[SPLAT].length > 0 ? JSON.stringify(info[SPLAT]) : '';
-            return `[${info.level.toUpperCase()}]${info['service'] ? ` [${info['service']}] ` : ''}${info.message} ${splat}`;
+            return `[${info.level.toUpperCase()}] ${instanceId}${info['service'] ? ` [${info['service']}] ` : ''}${info.message} ${splat}`;
```
Comment (TBonnin) [inline packages/utils/lib/logger.ts]: does this add an extra space when not in cloud but no space after when cloud?

Comment (nalanj) [reply packages/utils/lib/logger.ts]: Oh good catch, I'll fix that.

**Comment by nalanj** [comment]
@TBonnin here's how it looks running on my computer with `NANGO_CLOUD` set:

image

File: packages/utils/lib/logger.ts (hunk 2 above)
Comment (nalanj) [reply packages/utils/lib/logger.ts]: Fixed in aac9eb623

File: packages/utils/lib/logger.ts (hunk 1 above)
Comment (nalanj) [reply packages/utils/lib/logger.ts]: Set to 6 in aac9eb623

File: packages/utils/lib/logger.ts [hunk 3]
```diff
@@ -41,10 +42,12 @@ if (!isCloud && !isEnterprise) {
         })
     ];
 } else {
+    const instanceId = isCloud ? ` ${nanoid(6)}` : '';
+
     formatters = [
         winston.format.printf((info) => {
             const splat = info[SPLAT] && info[SPLAT].length > 0 ? JSON.stringify(info[SPLAT]) : '';
-            return `[${info.level.toUpperCase()}]${info['service'] ? ` [${info['service']}] ` : ''}${info.message} ${splat}`;
+            return `[${info.level.toUpperCase()}]${instanceId}${info['service'] ? ` [${info['service']}] ` : ''}${info.message} ${splat}`;
```
Comment (bodinsamuel) [inline packages/utils/lib/logger.ts:50]: You have access to `RENDER_INSTANCE_ID` that does exactly what you want 
https://render.com/docs/environment-variables#all-runtimes

**Review by bodinsamuel — APPROVED** [review]

File: packages/utils/lib/logger.ts (hunk 3 above)
Comment (nalanj) [reply packages/utils/lib/logger.ts:50]: It's just the hostname, which is `srv-cfifsvha6gdq1k7u46a0-99959bc5b-6mcdc` on one of our current instances. Not sure if we want to dump something like that into every log message we have vs. a 6 character identifier.

**Review by bodinsamuel — APPROVED** [review]

**Pull request merged**